### ISSUE TITLE
Configuration API exposes passwords (or password hashes)

### ISSUE THREAD

**Issue by voutilad** [attivio/searchui#4]
If using XML as the authentication mechanism, passwords are returned in the resulting call to the `/configuration` endpoint.

The default servlet configuration exposes `/users` publicly without any required authentication.

Even if deferring calls to `/users` after authenticated, this will leak passwords of other users if using XML to configure users and passwords.

**Comment by voutilad** [comment]
On closer inspection, nobody should use XML authentication in production as it seems to make the client-side authenticate the user. The docs should be updated to reflect that.

I'd caution that nobody should use XML authentication in a standalone SearchUI deployment until authentication is handled server-side.

**Comment by luigivaldez** [comment]
I'm sorry, I thought we had documented that people shouldn't use the XML-based authentication in production settings, at least for now. We will fix that. (I've added an issue — https://github.com/attivio/searchui/issues/5 — to do that.)

The XML auth itself ended up being a non-requirement for the initial release. In the future, we could add it as an option in the Spring security configuration of the servlet if it is something people end up wanting.
